### ISSUE TITLE
Deque's Axe Plugin catches WCAG violations in base storybook boilerplate

### ISSUE THREAD

**Issue by angel1199410** [storybookjs/storybook#11062]
**Describe the bug**
When running Deque's Axe plugin on a webpage, I found multiple accessibility errors with the Storybook boilerplate.

**To Reproduce**
1. Load and run the Storybook boilerplate with generic components - Button with Text and Button with Emoji.
2. Download [Deque's Axe Google Chrome plugin](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd/related?hl=en-US)
3. Run the plugin on the running boilerplate. 
4. See accessibility errors with exact selectors where issues were found.

**Expected behavior**
Axe should return no errors.  

**System:**
Binaries:
Node: 10.15.3
npm: 6.4.1
npmPackages:
@storybook/cli: 5.3.17
@storybook/html: 5.3.17

**Additional context**
Some issues have already been opened with solutions to the problems within Storybook's boilerplate.  I've linked to those, where applicable.  I have also linked directly to Deque's documentation on the required fixes for each accessibility defect.

1. [Page must have means to bypass repeated blocks.
](https://dequeuniversity.com/rules/axe/3.0/bypass)
    - Previously-opened issue with this same problem: https://github.com/storybookjs/storybook/issues/4667

2. [Page must have one main landmark.
](https://dequeuniversity.com/rules/axe/3.3/landmark-one-main#:~:text=It%20is%20a%20best%20practice%20to%20ensure%20that%20there%20is,or%20just%20a%20single%20landmark)

3. [Page must contain a level-one heading.
](https://dequeuniversity.com/rules/axe/3.5/page-has-heading-one)

4. [All page content must be contained by landmarks.
](https://dequeuniversity.com/rules/axe/3.2/region)

5. [Text Elements must have sufficient color contrast against the background](https://dequeuniversity.com/rules/axe/3.5/color-contrast?application=msftAI)
    - Previously-opened issue with this same problem: https://github.com/storybookjs/storybook/issues/8935

Running the plugin before every release will help validate that no accessibility bugs exist and that consumers are compliant when building out storybook applications.  Thank you for your time!

**Comment by shilman** [comment]
@yannbf @tooppaaa i know you've got a lot on your plate with the CLI overhaul but FYI....

**Comment by adagar** [comment]
@shilman I work full-time in A11Y, and have been looking for some open-source to contribute to. Could I take this ticket on?

**Comment by yannbf** [comment]
Hey @adagar that sounds incredible! Feel free to go ahead and if you ever need any direction in the repo, let me know!

**Comment by adagar** [comment]
Thanks, @yannbf I'm cracking it open now. I'll let you know if I run into trouble!

**Comment by adagar** [comment]
@yannbf was digging in some more on this; and I think I could use a bit more of an orientation on the repo as a whole. In particular, I'm curious if you're thinking of splitting this issue up into sub-issues (like items 1 & 5 have) and/or if we wanted to fix some of these as part of this Issue ticket. I think some guidance on what the landmarks we'd expect to be in use would also be handy. 

Do you have some time to hop on a Zoom call and discuss this week?

**Comment by adagar** [comment]
So results of initial investigations are basically that this repo needs to be updated to move towards semantic elements and rely less on `<div>`s. I'm experimenting a bit with some updates, but I'll def want to get some time with someone with more experience in this project.

**Comment by yannbf** [comment]
Awesome @adagar!! I really appreciate the time you're putting in this.

For more structural info @shilman is definitely the guy. I personally think that splitting the work in sub-issues will help things get moving, and also give opportunity for more contributors to join if they'd like as well as allow us to reach better a11y in steps (maybe by critical prio).

This week I'm quite busy unfortunately, maybe you can check with @shilman, but overall you're free to reach us out in the [discord community](https://discord.gg/h5rN2Bg2). Feel free to ping or DM!

**Comment by shilman** [comment]
Hey @adagar ! DM me on discord--I'd love to talk it through

**Comment by adagar** [comment]
Ticket has been created for issue #2 https://github.com/storybookjs/storybook/issues/13794

**Comment by adagar** [comment]
Regarding issue #3 : I don't see this issue flagged by Axe, but this may be because I have an `h1` element that seems to be an error
![Screen Shot 2021-02-02 at 4 06 15 PM](https://user-images.githubusercontent.com/3698858/106662639-a42bab80-6570-11eb-853f-b28fd85a2195.png)

This header isn't visible on-screen, and it's also probably not what you'd consider the most crucial piece of information on the page. 

I created a followup ticket to address that issue: https://github.com/storybookjs/storybook/issues/13796

**Comment by adagar** [comment]
An additional Axe violation that wasn't caught in the initial ticket has a new request: https://github.com/storybookjs/storybook/issues/13797

**Comment by adagar** [comment]
Created a PR for issue #2 https://github.com/storybookjs/storybook/pull/13827
